components/developer/golang-18/Makefile

```diff
@@ -0,0 +1,159 @@
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright (c) 2014, 2015, Oracle and/or its affiliates. All rights reserved.
+# Modified by Aurelien Larcher, 2016.$
+# Modifies by Till Wegmüller, 2017
+#
+
+include ../../../make-rules/shared-macros.mk
+
+PATCH_EACH_ARCHIVE= 1
+
+COMPONENT_NAME=			golang
+COMPONENT_VERSION=		1.8.3
+COMPONENT_REVISION=		0
+COMPONENT_MAJOR_VERSION=	$(shell echo $(COMPONENT_VERSION) | gsed -e 's/\([0-9]\+\.[0-9]\+\).*/\1/')
+COMPONENT_FMRI=			developer/golang
+COMPONENT_CLASSIFICATION=	Development/Other Languages
+COMPONENT_SUMMARY=		The Go Programming Language
+COMPONENT_SRC=			$(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_PROJECT_URL=		https://golang.org/
+COMPONENT_ARCHIVE=		go$(COMPONENT_VERSION).src.tar.gz
+COMPONENT_ARCHIVE_URL=		https://storage.googleapis.com/golang/$(COMPONENT_ARCHIVE)
+COMPONENT_ARCHIVE_HASH=		sha256:5f5dea2447e7dcfdc50fa6b94c512e58bfba5673c039259fd843f68829d99fa6
+COMPONENT_LICENSE=		BSD-style, Patent Grant
+COMPONENT_LICENSE_FILE=		$(COMPONENT_NAME).license
+
+# Go has a number of supplementary tools and packages that are located in
+# separate archives or source repositories. Because Go currently forces a
+# recompile of all packages whenever the compiler changes, building and
+# delivering them within the same build component build process is the best
+# option for now.
+
+# This version of Go is not actually packaged; it is only used to bootstrap the
+# actual version of the Go that will be packaged.  The long-term plan is
+# roughly as follows:
+#
+# - verify 1.5 produces "correct" result for bootstrapping 1.5 (upstream
+#   still uses 1.4 officially)
+# - after 1.5 has been available for at least a few builds in the WOS, switch
+#   to building Go with system version of Go
+# - when SPARC support is integrated into Go upstream, build upstream version
+#   using system Go for x86, then build Go for SPARC using upstream version
+#   just built for x86 (cross-compilation), package both at the same time
+# - after SPARC version has been available for at least a few builds in the
+#   WOS, switch to building Go with system version for x86 and SPARC
```

Comment:
Yes I am still using 1.4 to bootstrap 1.8 compilation. I could look into upstream userland to see if that has changed yet but for now a simple version upgrade and some. Best ofcourse would be an internal boostrapper in go 1.8 but I have not checked if that is the case yet. But this works and it's not much of a delay. It still compiles faster than any C or C++ thus far.